src/impl/meta_utils.hpp

```diff
@@ -35,24 +44,59 @@ struct filter_impl<Condition, Head, Tail...> {
                  >::type;
 };
 
-}  // namespace
+template <template <typename> class Condition, typename... Ts>
+using filter = typename filter_impl<Condition, Ts...>::type;
 
-namespace graph {
-class GraphTrait;
-class EdgeTrait;
-}  // graph
+template <typename, typename...>
+struct replace_abstract_traits_impl;
 
-namespace graph_impl {
+template <typename Node>
+struct replace_abstract_traits_impl<Node> {
+    using type = list<>;
+};
 
-template <template <typename> class Condition, typename... Ts>
-using filter = typename filter_impl<Condition, Ts...>::type;
+template <typename Node, typename Head, typename... Tail>
+struct replace_abstract_traits_impl<Node, Head, Tail...> {
```

Comment:
This cannot be used as a customization point by the user.

Reply:
It shouldn't be.
User should inherit from `GraphTrait` and `EdgeTrait` for customization.

Reply:
Ok, but how a user will do the same trick with its own trait (abstract -> impl)?

Reply:
I guess user will use `CustomTrait<CustomNode>` and it should be ok.
I'll think about it.